sermant-agentcore/sermant-agentcore-implement/src/main/java/io/sermant/implement/service/dynamicconfig/nacos/NacosClient.java

```diff
@@ -352,12 +355,24 @@ private String buildUrl(String key, String group, String namespace, boolean exac
     private String getToken() {
         if ((System.currentTimeMillis() - lastRefreshTime) >= TimeUnit.SECONDS
                 .toMillis(tokenTtl - TOKEN_REFRESH_WINDOW)) {
-            lastRefreshTime = System.currentTimeMillis();
-            HttpLoginProcessor httpLoginProcessor = new HttpLoginProcessor(
-                    NamingHttpClientManager.getInstance().getNacosRestTemplate());
-            LoginIdentityContext loginIdentityContext = httpLoginProcessor.getResponse(properties);
-            lastToken = loginIdentityContext.getParameter(KEY_ACCESS_TOKEN);
-            tokenTtl = Long.parseLong(loginIdentityContext.getParameter(KEY_TOKEN_TTL));
+            final StringBuilder requestUrl = new StringBuilder().append(HTTP_PROTOCOL);
+            requestUrl.append(properties.getProperty(PropertyKeyConst.SERVER_ADDR))
+                    .append(LOGIN_URL);
+            Map<String, String> formParams = new HashMap<>();
+            formParams.put(PropertyKeyConst.USERNAME, properties.getProperty(PropertyKeyConst.USERNAME));
+            formParams.put(PropertyKeyConst.PASSWORD, properties.getProperty(PropertyKeyConst.PASSWORD));
+            try {
+                String result = doPost(requestUrl.toString(), formParams, false);
+                if (StringUtils.isBlank(result)) {
+                    throw new RuntimeException("Nacos http request getToken exception, response is blank.");
+                }
+                JSONObject jsonObject = JSONObject.parseObject(result);
+                lastToken = jsonObject.getString(KEY_ACCESS_TOKEN);
+                tokenTtl = jsonObject.getLong(KEY_TOKEN_TTL);
+                lastRefreshTime = System.currentTimeMillis();
```

Comment:
Set the lastRefreshTime  before calling to ensure that the client token has expired when the server token expires.